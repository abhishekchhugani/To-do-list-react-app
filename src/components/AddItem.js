//Component to dispaly input box used to add a to do item


import uuid from 'uuid';
import React, { Component } from 'react'

class AddItem extends Component {
    state = {
        content: ''
    }
    handleChange = (e) => {
        this.setState({
            content: e.target.value
        });
    }
    handleSubmit = (e) => {
        e.preventDefault();
        // function call to add a to do item
        this.props.addTodoItem({id: uuid(), content:this.state.content, complete: false});
        this.setState({
            content: ''
        })
    }
    render() {
        return (
            <div className="row">
                <form className="col m12">
                    <div className="row mb0">
                        <div className="input-field input-group col s11">
                            <input id="icon_prefix" type="text" placeholder="Add Todo Item" className="validate" onChange={this.handleChange} value={this.state.content} />
                        </div>
                        <div className="input-field input-group col s1">
                            <span className="right">
                                <a className="btn btn-floating waves-effect waves-light red" onClick={this.handleSubmit}>+</a>
                            </span>
                        </div>
                    </div>
                </form>
            </div>
        )
    }
}

export default AddItem;
