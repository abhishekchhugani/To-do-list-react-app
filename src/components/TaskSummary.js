//Component to display count of total to do items and completed to do items

import React from 'react';

const TaskSummary = ({ todos }) => {

    const todo = todos.length ? (
        todos.filter(todo => {
            return todo.complete;
        })
    ) : []
    
    return (
        <div className="task-summary mb20">
            <div className="row inner-height">
                <div className="col s12 valign-wrapper horizontal-center total" id="totalCount">
                    Total: {todos.length}
                </div>
                <div className="col s12 valign-wrapper horizontal-center" id="completedCount">
                    Completed : {todo.length}
                </div>
            </div>
        </div>
    )
}

export default TaskSummary;


