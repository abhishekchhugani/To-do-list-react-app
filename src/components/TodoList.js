//Component to display the list of todo items

import React from 'react';

const TodoList = ({ todos, deleteTodo, markComplete, sortItems }) => {
  const todoList = todos.length ? (
    todos.map(todo => {
      return (
        <li className="collection-item avatar valign-wrapper">
          {
            todo.complete ? <i className="material-icons circle green">done</i> : <i className="material-icons circle blue">list</i>
          }
          <span className="title">{todo.content}</span>
          <span className="secondary-content mt10">
            <a className="waves-effect waves-light btn-small mr40" onClick={() => { markComplete(todo.id) }}><i className="material-icons left">cloud_done</i>Mark as done</a>
            <a href="#!" className="waves-effect waves-light mr10" onClick={() => { deleteTodo(todo.id) }}><i className="material-icons red-text">delete</i></a>
          </span>
        </li>
      )
    })
  ) : (
      <p className="center">Hurray! you have no outstanding tasks to be completed.</p>
    );

  return (
    <div>
      {
        todos.length > 1 ?
        <div className="sort-options">
          <div className="sortByName right-align mb20">
            <a class="waves-effect waves-light btn-large" onClick={sortItems()}>Sort by name</a>
          </div>
        </div> : void(0)
      }
      <ul className="collection">
        {todoList}
      </ul>
    </div>

  )
}

export default TodoList;


