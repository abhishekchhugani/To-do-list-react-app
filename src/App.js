// This is the main app container where the page loads

import React, { Component } from 'react';
import AppHeader from './components/AppHeader'
import AddItem from './components/AddItem'
import TodoList from './components/TodoList'
import TaskSummary from './components/TaskSummary'
import index from './index.css'

class App extends Component {
  state = {
    todos: []
  }
  //function to delete a to do item
  deleteTodoItem = (id) => {
    const todos = this.state.todos.filter(todo => {
      return todo.id !== id
    });
    this.setState({
      todos
    });
  }

  //Function to mark a to do item complete
  markComplete = (id) => {
    const todos = this.state.todos
    todos.map(todo => {
      todo.complete = todo.id === id ? true : todo.complete
    });
    this.setState({
      todos
    });
  }

  //Function to add a new to do item
  addTodoItem = (todoItem) => {
    let todos = this.state.todos
    todos.push(todoItem)
    this.setState({
      todos
    });
  }

  //function to sort the to do item
  sortItems = () => {
    let todos = this.state.todos
    todos.sort(function (a, b) {
      var nameA = a.content.toLowerCase(), nameB = b.content.toLowerCase();
      if (nameA < nameB)
        return -1;
      if (nameA > nameB)
        return 1;
      return 0;
    });
    this.setState({
      todos
    });
  }

  render() {
    return (
      <div className="todo-app">
        <AppHeader />
        <div className="container mt40 center">
          <TaskSummary todos={this.state.todos} />
          <AddItem addTodoItem={this.addTodoItem} />
          <TodoList todos={this.state.todos} deleteTodo={this.deleteTodoItem} markComplete={this.markComplete} sortItems={() => this.sortItems}/>
        </div>
      </div>
    );
  }
}

export default App;
