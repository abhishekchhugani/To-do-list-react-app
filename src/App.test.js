import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, exists } from 'enzyme';
import App from './App';
import TaskSummary from './components/TaskSummary'
import AddItem from './components/AddItem'
import TodoList from './components/TodoList'




  const todos = [
    {
        id: 1,
        content: 'Hello',
        complete: true
    },
    {
        id: 2,
        content: 'World',
        complete: false
    }
]

it('renders without crashing', () => {
    shallow(<App />);
});

it('snapshot test', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
});


it('Displays Task Summary', () => {
    const wrapper = mount(<App todos={todos}/>);
    expect(wrapper.find(<TaskSummary></TaskSummary>))
});

it('Displays Task Summary', () => {
    const wrapper = mount(<App todos={todos}/>);
    expect(wrapper.find(<AddItem></AddItem>))
});

it('Displays Task Summary', () => {
    const wrapper = mount(<App todos={todos}/>);
    expect(wrapper.find(<TodoList></TodoList>))
});
