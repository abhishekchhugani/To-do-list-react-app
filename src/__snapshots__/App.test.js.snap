// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AppHeader />,
        <div
          className="container mt40 center"
>
          <TaskSummary
                    todos={Array []}
          />
          <AddItem
                    addTodoItem={[Function]}
          />
          <TodoList
                    deleteTodo={[Function]}
                    markComplete={[Function]}
                    sortItems={[Function]}
                    todos={Array []}
          />
</div>,
      ],
      "className": "todo-app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TaskSummary
              todos={Array []}
/>,
            <AddItem
              addTodoItem={[Function]}
/>,
            <TodoList
              deleteTodo={[Function]}
              markComplete={[Function]}
              sortItems={[Function]}
              todos={Array []}
/>,
          ],
          "className": "container mt40 center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "todos": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addTodoItem": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "deleteTodo": [Function],
              "markComplete": [Function],
              "sortItems": [Function],
              "todos": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AppHeader />,
          <div
            className="container mt40 center"
>
            <TaskSummary
                        todos={Array []}
            />
            <AddItem
                        addTodoItem={[Function]}
            />
            <TodoList
                        deleteTodo={[Function]}
                        markComplete={[Function]}
                        sortItems={[Function]}
                        todos={Array []}
            />
</div>,
        ],
        "className": "todo-app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TaskSummary
                todos={Array []}
/>,
              <AddItem
                addTodoItem={[Function]}
/>,
              <TodoList
                deleteTodo={[Function]}
                markComplete={[Function]}
                sortItems={[Function]}
                todos={Array []}
/>,
            ],
            "className": "container mt40 center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "todos": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addTodoItem": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "deleteTodo": [Function],
                "markComplete": [Function],
                "sortItems": [Function],
                "todos": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
